### PR TITLE
Random test questions! 

### PR DESCRIPTION
**Pull request by LFGSaito**
Solves #1

### CODE REVIEW

File: qrmbot.py [hunk 1]
```diff
@@ -119,6 +120,45 @@
...
+    if level == "general":
+        rq = "General.json"
+    if level == "extra":
+        rq = "Extra.json"
+    if level is None:
+          list = ['Tech.json','General.json','Extra.json']
+          rq = random.choice(list)
+    try:
+        with open(rq) as data_file:
+            HQS = json.load(data_file)
```
Comment (galengold) [inline qrmbot.py]: can you load all 3 JSON files in the setup so it doesn't do it each time the command is selected?

Comment (galengold) [reply qrmbot.py]: the setup is below the ###########################

File: qrmbot.py [hunk 2]
```diff
@@ -119,6 +120,45 @@
...
+                            "\n**B:** "+question["answers"][1] +
+                            "\n**C:** "+question["answers"][2] +
+                            "\n**D:** "+question["answers"][3], inline=False)
+    embed = embed.add_field(name="Answer:", value="Type _?rqa_ for answer", inline=False)
+    global lastq
+    lastq[ctx.message.channel.id] = question["answer"]
+    await bot.say(embed=embed)
+
+@bot.command(pass_context=True)
+async def rqa(ctx):
```
Comment (galengold) [inline qrmbot.py]: Please add a docstring so help text will be generated for this command

File: qrmbot.py [hunk 3]
```diff
@@ -119,6 +120,45 @@
...
+    if level is None:
+          list = ['Tech.json','General.json','Extra.json']
+          rq = random.choice(list)
+    try:
+        with open(rq) as data_file:
+            HQS = json.load(data_file)
+            question = random.choice(HQS)
+            #question = HQS[0]
+    except FileNotFoundError:
+         await bot.say("I can't find the level you told me. Try typeing either ?rq, ?rq tech, ?rq general, ?rq extra.")
```
Comment (galengold) [inline qrmbot.py]: maybe allow tech, general, extra, t, g, e, and all as arguments (all is the same as no argument)?

File: qrmbot.py [hunk 4]
```diff
@@ -119,6 +120,45 @@
     '''Posts an image of the US Frequency Allocations'''
     await bot.send_file(ctx.message.channel, 'band-chart.png')
 
+@bot.command(aliases=['randomq'], pass_context=True)
+async def rq(ctx, level : str = None):
+    '''Gets a random question from the Technian/General/Extra question pools'''
+    rq = ""
+
+    if level == "tech":
```
Comment (galengold) [inline qrmbot.py]: make it easier to invoke correctly be adding `level = level.lower()`

File: qrmbot.py (hunk 2 above)
Comment (LFGSaito) [reply qrmbot.py]: Fixed.

File: qrmbot.py (hunk 4 above)
Comment (LFGSaito) [reply qrmbot.py]: Fixed.

File: qrmbot.py (hunk 1 above)
Comment (LFGSaito) [reply qrmbot.py]: I've tried a few ways of doing this and I've had nothing but error. If you know of a better way feel free to make a pull request on my repo.

File: qrmbot.py (hunk 3 above)
Comment (LFGSaito) [reply qrmbot.py]: Fixed. All didn't want to work but bot will help people out. To be commited soon.

File: qrmbot.py (hunk 1 above)
Comment (galengold) [reply qrmbot.py]: make 3 json objects with `json.load` during the setup in a list\* then do random.choice from the list, or choose the pool from the list.

\* sidenote: don't name a variable `list` like on line 145, `list` is the name of the constructor function for lists (`list()`), and can cause unintended behaviors.

**Comment by ntoskrnl4** [comment]
The changes I made should address Galen's requested changes, but I did not test that it works because I use the rewrite version of discord.py myself.

**Pull request merged**